base/reduce.jl

```diff
@@ -422,7 +478,7 @@ julia> prod(1:20)
 2432902008176640000
 ```
 """
-prod(a) = mapreduce(promote_sys_size_mul, *, a)
+prod(a) = mapreduce(identity, mul_prod, a)
 
 ## maximum & minimum
 
```

Comment:
It might be worth changing the `f(a1)` in the below function to `mapreduce_single(f, op, a1)` for consistency.

Reply:
done.